packages/graph/lib/graph.ts

```diff
@@ -7,6 +8,8 @@ import { Node } from "./node";
  * Graph represents a
  */
 export class Graph {
+    public nodes_list: string[] = [];
```

Comment:
Probably don't node `nodes_list`.  Can easily iterate over a Map type:

```javascript
const map = new Map();
map.set(1, 'a');
map.set(2, 'b');
map.set(3, 'c');

// iterate over key-values since Map is an iterator
for(const [key, val] of map) {
   console.log(key, val);
}

// iterate over keys as an iterator
for(const key of map.keys()) {
   console.log(key);
}

// iterate over vals as an iterator
for(const val of map.values()) {
   console.log(val);
}
```  

Lastly you can covert any iterator into an Array with `Array.from`:

```javascript
console.log(Array.from(map));
console.log(Array.from(map.keys()));
console.log(Array.from(map.values()));
```

Reply:
Sure thing will work with maps (;